webpack.config.js

```diff
@@ -117,7 +125,15 @@ const webpackConfig = {
 		new webpack.ProvidePlugin( {
 			process: 'process/browser',
 		} ),
-	],
+		/**
+		 * If the development environment uses HTTPS,
+		 * it will fail when first connecting to webpack dev server,
+		 * so turn off the `overlay` option here.
+		 * Ref: https://github.com/pmmmwh/react-refresh-webpack-plugin/blob/v0.5.4/docs/TROUBLESHOOTING.md#component-not-updating-with-bundle-splitting-techniques
+		 */
+		hasReactFastRefresh &&
+			new ReactRefreshWebpackPlugin( { overlay: false } ),
+	].filter( Boolean ),
```

Comment:
💅 

I'd add a comment somewhere clarifying this filter(Boolean) is for removing falsy items because maybe `ReactRefreshWebpackPlugin` is not loaded.

 Notice other option could be to use spread. 
 
 ```
requestToHandle,
} ),
/**
		 * Automatic polyfills for native node.js modules were removed from webpack v5.
		 * And `@wordpress/components` below version 16.x depends on `@wordpress/compose`,
		 * which directly accesses `process.env`.
		 * Ref: https://github.com/WordPress/gutenberg/blob/%40wordpress/components%4012.0.8/packages/compose/src/hooks/use-reduced-motion/index.js#L21
		 *
		 * So the fallback is required here.
		 * It may be possible to remove this fallback
		 * when `@wordpress/components` is upgraded above 17.0.0+,
		 * or when it is removed from the `requestToExternal` function above.
		 */
...(hasReactFastRefresh ? [] : [new ReactRefreshWebpackPlugin( { overlay: false } ) ])
 ],
 ``` 

Or even more reusable

```
requestToHandle,
} ),
...reactFastRefreshPlugins
]
```

```
const reactFastRefreshPlugins = () => {
    if (! hasReactFastRefresh) return [];
    
    // Enabled plugins with hasReactFastRefresh go here
    return [
        new ReactRefreshWebpackPlugin( { overlay: false } ),
    ]
}
```

Reply:
I took another way in https://github.com/woocommerce/google-listings-and-ads/pull/1345/commits/7421728b46738229945c010ee5ba12a891c43e5f. Please have a look.

Reply:
Looking good, thanks